plugins/module_utils/validation.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) IBM Corporation 2023
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+"""
+Class implemented for common validations that are not specific to z/OS but rather system or
+security related.
+
+"""
+import os
+
+
+def validate_safe_path(path):
+    """
+    This function is implemented to validate against path traversal attack
+    when using os.path.join function.
+
+    In this action plugin, path is on the controller.
+    """
+    if not os.path.isabs(path):
+        real_path = os.path.realpath(path)
+        if not os.path.exists(real_path) and not real_path.endswith(os.sep):
```

Comment:
This is a good assumption, took me a couple of minutes , comment was helpful, for others, if the path does not exist its likely as stated a data set, while we could go and start calling data set exists and other data set validation calls, there are those cases where a data set is passed to us to be created by us thus we would ultimately end up return path anyhow because we do create data sets from input. Nice.